### PR TITLE
End-End Tests

### PR DESCRIPTION
**Pull request by tkatzir**
Adding capabilities for TestSuite

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://jira-hq.paloaltonetworks.local/browse/CIAC-870

### CODE REVIEW

File: TestSuite/text_based.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def __init__(self, dir_path: Path, name: str):
         self.write_text()
 
     def write_text(self, text: str = ''):
-        self._file_path.write_text(text)
+        self._file_path.write_text(data=text, encoding=None)
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply TestSuite/text_based.py:13]: It fails otherwise. The encoding seems to be a mandatory param.

File: tests_end_to_end/e2e_tests_utils.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    git.Repo.clone_from(url='https://github.com/demisto/demisto-sdk.git',
+                        to_path=destination_folder,
+                        multi_options=[
+                            f'-b {sdk_git_branch}',
+                            '--single-branch',
+                            '--depth 1'
+                        ]
+                        )
+
+    sys.path.insert(1, f'{destination_folder}')
```
Comment (dorschw) [inline tests_end_to_end/e2e_tests_utils.py:23]: is this necessary now that the e2e tests are part of the SDK?

File: tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+from demisto_sdk.commands.format.format_module import format_manager
+from demisto_sdk.commands.generate_docs import generate_playbook_doc
+from demisto_sdk.commands.upload.uploader import Uploader
+from demisto_sdk.commands.validate.validate_manager import ValidateManager
+
+
+def test_e2e_demisto_sdk_flow_playbook_testsuite(tmpdir):
+    # Importing TestSuite classes from Demisto-SDK, as they are excluded when pip installing the SDK.
+    e2e_tests_utils.cli(f'mkdir {tmpdir}/git')
+    e2e_tests_utils.git_clone_demisto_sdk(destination_folder=f'{tmpdir}/git/demisto-sdk', sdk_git_branch='testsuite-playbook')
```
Comment (dorschw) [inline tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py:17]: see comment above

File: tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    from TestSuite.playbook import Playbook
+    from TestSuite.repo import Repo
+
+    repo = Repo(tmpdir)
+
+    unique_id = random.randint(1, 1000)
+    pack_name = 'foo_' + str(unique_id)
```
Comment (dorschw) [inline tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py:24]: create_pack adds a serial number automatically afaik

File: tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                    "task": {}
+                }
+            }
+        }
+    ]
+
+    header_params = {}
+    header_params['Accept'] = 'application/json'  # noqa: E501
+    header_params['Accept-Encoding'] = 'gzip, deflate, br'
+    header_params['Content-Type'] = 'application/json'
```
Comment (dorschw) [inline tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py:83]: we can do `header_params = { a:a_value , b:b_value }` instead

**Review by dorschw — COMMENTED** [review]
Nice

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,6 +26,8 @@
 * Added the `DEMISTO_SDK_CONTENT_PATH` environment variable, implicitly used in various commands.
 * Added link to documentation for error messages regarding use cases and tags.
 
+* Added playbook modification capabilities in **TestSuite**.
```
[the remark this answers is not in the document]
Comment (tkatzir) [reply CHANGELOG.md]: [Done](https://github.com/demisto/demisto-sdk/pull/2269/commits/a746b3a417f577aac06817c3f14e0b4ac9a93894).

File: tests_end_to_end/e2e_tests_utils.py (hunk 2 above)
Comment (tkatzir) [reply tests_end_to_end/e2e_tests_utils.py:23]: Yes. It's run from the Content's sdk-nightly, and TestSuite is excluded when `pip instal`ling the sdk.

File: tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py (hunk 3 above)
Comment (tkatzir) [reply tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py:17]: Answered above. :-)

File: tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py (hunk 4 above)
Comment (tkatzir) [reply tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py:24]: I'm (re)using the same folder. Without this - I couldn't do successive runs.

File: tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py (hunk 5 above)
Comment (tkatzir) [reply tests_end_to_end/test_e2e_demisto_sdk_flow_playbook.py]: [Done](https://github.com/demisto/demisto-sdk/pull/2269/commits/28de6af8cb2b1b6574220d56f323fe802c3e34b4).

**Review by dorschw — APPROVED** [review]

**Pull request merged**